medusa/orchestration.py

```diff
@@ -60,7 +63,8 @@ def pssh_run(self, hosts, command, hosts_variables=None):
                                        cert_file=cert_file)
             logging.debug('Batch #{i}: Running "{command}" on nodes {hosts} parallelism of {pool_size}'
                           .format(i=i, command=command, hosts=parallel_hosts, pool_size=len(parallel_hosts)))
-            output = client.run_command(command, host_args=hosts_variables, sudo=True)
+            output = client.run_command(command, host_args=hosts_variables,
+                                        sudo=medusa.utils.evaluate_boolean(self.cassandra_config.use_sudo))
```

Comment:
The `cassandra_config` name is actually poorly chosen as it is a `config` object that you get here, not `config['cassandra']`:
 
```suggestion
                                        sudo=medusa.utils.evaluate_boolean(self.cassandra_config.cassandra.use_sudo))
```